adam-core/src/main/scala/org/bdgenomics/adam/hbase/HBaseFunctions.scala

```diff
@@ -0,0 +1,587 @@
+   * @param inConnection optional HBase Connection parameter, only used if manually configuring HBase connection
+   * @param inAdmin optional HBase Admin parameter, only used if manually configuring HBase connection
+   * @param inHBaseContext optional HBaseContext parameter, only used if manually configuring HBase connection
+   */
+
+  class HBaseSparkDAO(sc: SparkContext,
+                      inConf: Option[Configuration] = None,
+                      inConnection: Option[Connection] = None,
+                      inAdmin: Option[Admin] = None,
+                      inHBaseContext: Option[HBaseContext] = None) {
+
+    val conf = if (inConf.isDefined) inConf.get else HBaseConfiguration.create()
+    val connection = if (inConnection.isDefined) inConnection.get else ConnectionFactory.createConnection(conf)
+    val admin = if (inAdmin.isDefined) inAdmin.get else connection.getAdmin
+    val hbaseContext = if (inHBaseContext.isDefined) inHBaseContext.get else new HBaseContext(sc, conf)
+
+    def createTable(tableDescriptorMeta: HTableDescriptor) { admin.createTable(tableDescriptorMeta) }
+    def getTable(tableName: TableName): Table = { connection.getTable(tableName) }
+
+    def getHBaseRDD(tableName: org.apache.hadoop.hbase.TableName,
+                    scans: org.apache.hadoop.hbase.client.Scan): RDD[scala.Tuple2[org.apache.hadoop.hbase.io.ImmutableBytesWritable, org.apache.hadoop.hbase.client.Result]] = {
+      hbaseContext.hbaseRDD(tableName, scans)
+
+    }
+
+    def hbaseBulkLoad(genodata: RDD[(Array[Byte], List[(String, Array[Byte])])],
+                      hbaseTableName: String,
+                      flatMap: scala.Function1[(Array[Byte], List[(String, Array[Byte])]), scala.Iterator[scala.Tuple2[org.apache.hadoop.hbase.spark.KeyFamilyQualifier, scala.Array[scala.Byte]]]],
+                      stagingFolder: String) = {
+
+      val familyHBaseWriterOptions = new java.util.HashMap[Array[Byte], FamilyHFileWriteOptions]
+      val f1Options = new FamilyHFileWriteOptions("GZ", "ROW", 128, "FAST_DIFF")
+      familyHBaseWriterOptions.put(Bytes.toBytes("g"), f1Options)
+
+      // call implicit RDD function hbaseBulkLoad to write bulk files to staging locations
+      // and then load to HBase
+      genodata.hbaseBulkLoad(hbaseContext,
+        TableName.valueOf(hbaseTableName),
+        flatMap,
+        stagingFolder,
+        familyHBaseWriterOptions,
+        compactionExclude = false,
+        HConstants.DEFAULT_MAX_FILE_SIZE)
+
+      // This permission change appears necessary, plan to revisit to find better way
+      ("hadoop fs -chmod -R 660 " + stagingFolder) !
+
+      val load = new LoadIncrementalHFiles(conf)
+      load.doBulkLoad(new Path(stagingFolder), admin, connection.getTable(TableName.valueOf(hbaseTableName)),
+        connection.getRegionLocator(TableName.valueOf(hbaseTableName)))
+
+    }
+
+    def hbaseBulkDelete() {}
```

Comment:
Remove or fill with `???` so that a NotImplementedException gets thrown when it is called.